src/duplo_resource/configmap.py

```diff
@@ -102,22 +158,31 @@ def update(self,
       if 'metadata' not in body or body['metadata'].get('name') != name:
         raise DuploError("Provided 'name' must match 'metadata.name' in the body.")
     else:
-      body = {'metadata': {'name': name}, 'data': {}}
+     body = self.find(name)
+     if not body:
+        raise DuploError(f"ConfigMap '{name}' not found.")
+
     body.setdefault('data', {}).update(data or {})
-    return body if dryrun else super().update(body)
+    return body if dryrun else super().update(name=name, body=body, patches=patches)
 
   @Command()
   def find(self,
            name: args.NAME) -> dict:
-    """Find a ConfigMap by name and return it's content.
+    """Find a ConfigMap.
+
+    Retrieve details of a specific ConfigMap by name
+
     Usage: cli usage
       ```sh
       duploctl configmap find <name>
       ```
+
     Args:
       name: The name of the ConfigMap to find.
+
     Returns:
-      dict: The resource content or success message.
+      message: The resource content or success message.
+
     Raises:
       DuploError: If the ConfigMap could not be found.
     """
```

Comment:
Basically the the get will properly return the DuploError. It may not look like the best message. So if you want, put the raise DuploError here with a more specific message and remove the raise from the above. 
```suggestion
response = self.duplo.get(self.endpoint(name))
body = response.json()
if not body:
  raise DuploError(f"ConfigMap '{name}' not found.")
return body
```

Reply:
Done, I have catch the exception to send a good exception error specific to the resource.

Reply:
Yes, I do realize that now. You would have to try catch the get request.